app/_kong_plugins/json-threat-protection/index.md

```diff
@@ -38,4 +38,50 @@ search_aliases:
   - json-threat-protection
 ---
 
-## Overview
+The JSON Threat Protection plugin validates and protects against malicious or overly complex JSON payloads. This includes JSON-based injection attacks, oversized payload attacks, and malformed JSON leading to application crashes. 
+
+Enabling this plugin is recommended for any API that accepts JSON input, especially public APIs with user-submitted data and high-traffic APIs vulnerable to DDoS attacks using large payloads.
+
+## How it works
+
+The JSON Threat Protection plugin validates incoming requests with a JSON body against policy limits that you've configured for the plugin, regardless of whether the `Content-Type` header exists or is set to `application/json`. If a request violates the policy limits, you can configure it to either block the request (block mode) or monitor and log it (tap mode).
+
+The plugin checks the following limits:
+
+- Maximum container depth of the entire JSON
+- Maximum number of array elements
+- Maximum number of object entries
+- Maximum length of object keys
+- Maximum length of strings
+
+For example, for the following JSON:
+
+```json
+{
+  "name": "Jason",
+  "age": 20,
+  "gender": "male",
+  "parents": ["Joseph", "Viva"]
+}
+```
+
+- Maximum container depth: 2
+- Maximum number of array elements: 2
+- Maximum number of object entries: 4
+- Maximum length of object keys: 7 (`parents`)
+- Maximum length of strings: 6 (`Joseph`)
```

Comment:
The way this example is presented doesn't really make sense to me. It looks like we're calculating the maximum values for the existing JSON, not defining a limit.

I would start with a config example, and then have an example of JSON content that would be valid and another that would be against policy (and describe why). For example, if Maximum length of object keys is set to 6, then this JSON violates the policy because `parents` is over the limit.

Reply:
@lmilan I changed how this JSON violation example was formatted if you can take another look!